File: Day 5 /13.Roman to Integer.java
class Solution {
    public int getRoman(char ch) {
        switch (ch) {
            case 'I': return 1;
            case 'V': return 5;
            case 'X': return 10;
            case 'L': return 50;
            case 'C': return 100;
            case 'D': return 500;
            case 'M': return 1000;
            default: return -1; // invalid character
        }
    }

    public int romanToInt(String s) {
        int total = 0;
        int prev = 0;
        for (int i = s.length() - 1; i >= 0; i--) {  // RIGHT to LEFT loop
            char ch = s.charAt(i);
            int cur = getRoman(ch);
            if (cur < prev) {
                total -= cur;
            } else {
                total += cur;
            }
            prev = cur;
        }
        return total;
    }
}

File: Day 5 /13.Roman to Integer.cpp
class Solution {
public:
    int romanToInt(string s) {
        int values[256] = {0};
        values['I'] = 1;
        values['V'] = 5;
        values['X'] = 10;
        values['L'] = 50;
        values['C'] = 100;
        values['D'] = 500;
        values['M'] = 1000;
        
        int total = 0;
        int prev_value = 0;
        
        for (char c : s) {
            int current_value = values[c];
            if (current_value > prev_value) {
                total += current_value - 2 * prev_value;
            } else {
                total += current_value;
            }
            prev_value = current_value;
        }
        
        return total;
    }
};
